.github/workflows/publish-web-client-next.yml

```
@@ -1,68 +1,114 @@
-# This workflow publishes the 'web-client' package to npm if the version number in the 
-# 'web-client' package.json file has changed.
+# This workflow publishes the 'web-client' package to npm if a patch release label is applied to the PR and 
+# the 'web-client' package.json file has been updated.
 
 name: Publish Web Client SDK to NPM on Next
 
 on:
-  push:
+  pull_request:
     branches:
       - next
+    types:
+      - closed
 
 permissions:
   contents: read
 
 jobs:
   publish:
+    if: >
+      github.event.pull_request.merged == true &&
+      contains(github.event.pull_request.labels.*.name, 'patch release')
     runs-on: ubuntu-latest
 
     steps:
       - name: Check out code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
+          ref: next
 
       - name: Set up Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           node-version: '20'
           registry-url: 'https://registry.npmjs.org/'
 
-      - name: Set up Rust and wasm target
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          profile: minimal
-          override: true
-          target: wasm32-unknown-unknown
-          components: rust-src 
+      - name: Set up Rust and wasm target
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+          target: wasm32-unknown-unknown
+          components: rust-src
+
+      - name: Verify version bump against next
+        id: check_version
+        run: |
+          set -eo pipefail

```

nit: consider also adding the 'u' switch to avoid undefined vars
```suggestion
          set -euo pipefail
```